### PR TITLE
Send user weekly email using postmark

### PR DESCRIPTION
**Pull request by tfussell**
For email integration roadmap issue.

### CODE REVIEW

File: pkg/controller/queue/controller.go [hunk 1]
```diff
@@ -224,13 +225,17 @@ func (c *Controller) weekly(o func() error) error {
...
-			return nil
-		}
+		/*
+			if t.Weekday() != time.Monday {
+				return nil
+			}
+		*/
 
-		if m != 0 {
+		if m != 0 || h != 12 {
```
Comment (tfussell) [inline pkg/controller/queue/controller.go:238]: Temporarily (for testing) disabling emails only on Monday at 12AM and instead sending the email every day at noon (UTC).

File: pkg/controller/queue/controller.go [hunk 2]
```diff
@@ -260,7 +265,7 @@ func (c *Controller) weekly(o func() error) error {
 
 	{
 		val, err := c.redigo.Simple().Search().Value(k)
-		if err != nil {
+		if err != nil && !simple.IsNotFound(err) {
```
Comment (tfussell) [inline pkg/controller/queue/controller.go:268]: In the case where the key doesn't already exist in the DB, we'll never get to creating it unless we don't treat it as an error.

File: pkg/handler/remindercreate/user.go [hunk 3]
```diff
@@ -106,56 +107,107 @@ func (u *User) Filter(tsk *task.Task) bool {
...
 
 func (u *User) createReminder(tsk *task.Task) error {
-	var err error
+	uid := tsk.Obj.Metadata[metadata.UserID]
+	user, err := u.searchUser(uid)
+	if err != nil {
+		return err
+	}
+	userEmail := user.Obj.Property.Mail
+	emailFeatureEnabled := user.Obj.Metadata["feature.venturemark.co/weekly-update"] == "true"
```
Comment (tfussell) [inline pkg/handler/remindercreate/user.go:116]: Temporary per-user feature flag. Once we are confident in the feature we can drop this and enable it for all users.

File: pkg/handler/remindercreate/user.go [hunk 4]
```diff
@@ -106,56 +107,107 @@ func (u *User) Filter(tsk *task.Task) bool {
...
 
-	var ven []*schema.Venture
+	if !emailFeatureEnabled {
+		u.logger.Log(context.Background(), "level", "info", "message", "user has not opted in to weekly update emails, skipping", "user", uid)
+		return nil
+	} else if userEmail == "" {
```
Comment (tfussell) [inline pkg/handler/remindercreate/user.go:121]: User emails are not currently stored in the DB. To be safe, we will skip users that don't have an email while we are migrating them.

**Comment by tfussell** [comment]
Tested locally.

With flag on a user:
```
{ "caller":"github.com/venturemark/apiworker/pkg/handler/remindercreate/user.go:87", "level":"info", "message":"creating user reminder", "time":"2021-06-25 23:20:34", "user":"1624642229220519800" }
{ "caller":"github.com/venturemark/apiworker/pkg/handler/remindercreate/user.go:94", "level":"info", "message":"created user reminder", "time":"2021-06-25 23:20:35", "user":"1624642229220519800" }
```

Without flag:
```
{ "caller":"github.com/venturemark/apiworker/pkg/handler/remindercreate/user.go:87", "level":"info", "message":"creating user reminder", "time":"2021-06-25 23:02:22", "user":"1624642229220519800" }
{ "caller":"github.com/venturemark/apiworker/pkg/handler/remindercreate/user.go:119", "level":"info", "message":"user has not opted in to weekly update emails, skipping", "time":"2021-06-25 23:02:22", "user":"1624642229220519800" }
{ "caller":"github.com/venturemark/apiworker/pkg/handler/remindercreate/user.go:94", "level":"info", "message":"created user reminder", "time":"2021-06-25 23:02:22", "user":"1624642229220519800" }
```

**Review by marcusellison — APPROVED** [review]

File: pkg/handler/remindercreate/user.go [hunk 5]
```diff
@@ -106,56 +107,107 @@ func (u *User) Filter(tsk *task.Task) bool {
...
 			if err != nil {
 				return tracer.Mask(err)
 			}
 
-			for _, o := range u {
-				uid := o.Obj.Metadata[metadata.UpdateID]
-				dur := 168 * time.Hour
+			for _, currentUpdate := range timelineUpdates {
+				updateID := currentUpdate.Obj.Metadata[metadata.UpdateID]
+				dur := 24 /* * 7 */ * time.Hour
```
Comment (tfussell) [inline pkg/handler/remindercreate/user.go:158]: Changing to daily instead of weekly updates here temporarily.

**Pull request merged**